packages/amplify_api/lib/method_channel_api.dart

```diff
@@ -271,6 +271,11 @@ class AmplifyAPIMethodChannel extends AmplifyAPI {
   // ====== GENERAL METHODS ======
 
   ApiException _deserializeException(PlatformException e) {
+    // TEMP question: Would it also be possible/desirable to define more specific error
+    // exception classes here? For example, could we define HTTPNotFoundError (extends ApiError)
+    // and look at the serialized httpStatusCode to define which one to create.
+    // That way, customers would not need `code == 404` etc...
+
```

Comment:
callout here: this might be more work, but arguable a better DX. lmk thoughts

Reply:
IMO the DX improvement is minimal. I think it makes sense to get this in as is and then consider it for a future enhancement.

Reply:
Agreed, will revert change in this file.